cli/parser/parser.go

```diff
@@ -314,6 +314,35 @@ type CommandLineSchema struct {
 	// allow us to show plugin-specific help.
 }
 
+// CommandSupportsOpt returns whether the given opt is in the CommandOptions.
+// The opt is expected to be either "--NAME" or "-SHORTNAME", without an "=".
+func (s *CommandLineSchema) CommandSupportsOpt(opt string) bool {
+	// TODO: this func is using heuristics, since a correct impl would require
+	// us to know the schema for all bazel commands. At some point we should
+	// probably just do a one-time parse of all the commands so that we can do
+	// this properly, or see if Bazel can give us a dump of its option schema.
```

Comment:
Might be worth taking a look at https://github.com/bazelbuild/bazel/pull/19536#issuecomment-1793249509

It mentions a file called `[bazel_flags.proto](https://github.com/bazelbuild/bazel/pull/19536/files#diff-289cfc38fc9fe94347bca949249e42e12870d4d310d0c60c39cd266aba16f0b9)` that we could potentially use

Reply:
Ah looks like we can run `bazel help flags-as-proto` and it dumps a base64-encoded FlagCollection proto. Will save that for a future PR since it'll involve a pretty substantial rewrite.